diff --git a/net/tnftp/files/src/Makefile.in b/net/tnftp/files/src/Makefile.in
index 6ea18fb..99d2f95 100644
--- a/net/tnftp/files/src/Makefile.in
+++ b/net/tnftp/files/src/Makefile.in
@@ -20,6 +20,7 @@ CFLAGS	= @CFLAGS@
 CPPFLAGS= -I${srcdir} -I${srcdir}/.. -I. -I.. @INCLUDES@ @CPPFLAGS@
 LIBS	= @LIBS@
 LDFLAGS	= @LDFLAGS@
+CROSS_LDFLAGS = @LDFLAGS@
 
 INSTALL	= @INSTALL@
 
@@ -32,7 +33,7 @@ OBJS	= ${SRCS:.c=.o}
 all:	${PROG}
 
 ${PROG}: ${OBJS} @LIBDEPENDS@
-	${CC} ${CFLAGS} ${LDFLAGS} -o ${PROG} ${OBJS} ${LIBS}
+	${CC} ${CFLAGS} ${LDFLAGS} ${CROSS_LDFLAGS} -o ${PROG} ${OBJS} ${LIBS}
 
 ${OBJS}: ${srcdir}/../tnftp.h ../config.h
 
